src/main/kotlin/no/nav/familie/ef/sak/vedtak/VedtakService.kt

```diff
@@ -100,6 +119,62 @@ class VedtakService(
     fun hentHarAktivtVedtak(behandlingId: UUID, localDate: LocalDate = LocalDate.now()): Boolean {
         return hentVedtak(behandlingId).erVedtakAktivtForDato(localDate)
     }
+
+    @Transactional
+    fun angreSendTilBeslutter(saksbehandling: Saksbehandling) {
+        val vedtak = hentVedtak(behandlingId = saksbehandling.id)
+
+        validerKanAngreSendTilBeslutter(saksbehandling, vedtak)
+
+        behandlingService.oppdaterStegPåBehandling(saksbehandling.id, steg = StegType.SEND_TIL_BESLUTTER)
+        behandlingService.oppdaterStatusPåBehandling(saksbehandling.id, status = BehandlingStatus.UTREDES)
+        behandlingshistorikkService.opprettHistorikkInnslag(
+            behandlingId = saksbehandling.id,
+            stegtype = saksbehandling.steg,
+            utfall = StegUtfall.ANGRE_SEND_TIL_BESLUTTER,
+            metadata = null
+        )
+
+        ferdigstillGodkjenneVedtakOppgave(saksbehandling)
+        opprettBehandleSakOppgave(saksbehandling)
+    }
+
+    private fun opprettBehandleSakOppgave(saksbehandling: Saksbehandling) {
+        taskService.save(
+            OpprettOppgaveTask.opprettTask(
+                OpprettOppgaveTask.OpprettOppgaveTaskData(
+                    behandlingId = saksbehandling.id,
+                    oppgavetype = Oppgavetype.BehandleSak,
+                    beskrivelse = "Angret send til beslutter"
+                )
+            )
+        )
+    }
+
+    private fun ferdigstillGodkjenneVedtakOppgave(saksbehandling: Saksbehandling) {
+        oppgaveService.hentOppgaveSomIkkeErFerdigstilt(oppgavetype = Oppgavetype.GodkjenneVedtak, saksbehandling)?.let {
+            taskService.save(
+                FerdigstillOppgaveTask.opprettTask(
+                    behandlingId = saksbehandling.id,
+                    oppgavetype = Oppgavetype.GodkjenneVedtak,
+                    it.gsakOppgaveId,
+                    personIdent = null
+                )
+            )
+        }
+    }
+
+    private fun validerKanAngreSendTilBeslutter(saksbehandling: Saksbehandling, vedtak: Vedtak) {
```

Comment:
I denne kan vi nog bruke `brukerFeilHvis` på en del steder i stedet for `feilHvis`
Største forskjellen er at `feilHvis` logger error, som i de fleste tilfeller ikke er nødvendig, og at `brukerFeilHvis` har `HttpStatus.BAD_REQUEST` som default

Reply:
Hadde også vært fint med å justere formatteringen, med ny rad for feilmeldingen for å ikke få for lange rader
```kotlin
brukerfeilHvis(vedtak.saksbehandlerIdent != innloggetSaksbehandler) {
            "Kan ikke angre send til beslutter om du ikke er saksbehandler på vedtaket"
        }
```